server/data_updates/00015_20181121-145724_archive.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8; -*-
+# This file is part of Superdesk.
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+#
+# Author  : Jérôme
+# Creation: 2018-11-21 14:57
+
+from superdesk.commands.data_updates import DataUpdate
+
+
+class DataUpdate(DataUpdate):
+    """This migration add "translated_from" field to all items translated using manual duplication"""
+    resource = 'archive'
+
+    def forwards(self, collection, database):
```

Comment:
I see no trees here. `translation_id` has to be added to all child translations.

Reply:
@tomaskikutis what do you mean? It is added to all child translations.

Reply:
I mean if you create article A, translate it to B, then B to C, then C to D. They all must have the same `translation_id`. Does this script cover this?

Reply:
Yes, but I indeed realize there is a case where it can go wrong (if D is modified first, then B, we'll have 2 distinct `translation_id` which is bad). I'm fixing it, thanks for pointing this out.

Reply:
@tomaskikutis I've changed behaviour so the case I've mentioned above should not cause trouble anymore.

Reply:
Checked working with `A, translate it to B, then B to C, then C to D`